subworkflows/local/bin_qc.nf

```diff
@@ -81,57 +79,28 @@ workflow BIN_QC {
         /*
          * BUSCO
          */
+        busco_lineage = params.busco_auto_lineage_prok ? 'auto_prok' : 'auto'
+
         if (!ch_busco_db.isEmpty()) {
             if (ch_busco_db.extension in ['gz', 'tgz']) {
-                // Expects to be tar.gz!
-                BUSCO_DB_PREPARATION(ch_busco_db)
-                ch_db_for_busco = BUSCO_DB_PREPARATION.out.db.map { meta, db ->
-                    [[id: meta, lineage: 'Y'], db]
+                BUSCO_UNTAR([[id: ch_busco_db.getSimpleName()], ch_busco_db])
+
+                if (ch_busco_db.getSimpleName().contains('odb')) {
+                    busco_lineage = ch_busco_db.getSimpleName()
                 }
+                ch_busco_db = BUSCO_UNTAR.out.untar.map { it[1] }
             }
-            else if (ch_busco_db.isDirectory()) {
-                // Set meta to match expected channel cardinality for BUSCO
-                ch_db_for_busco = Channel
-                    .of(ch_busco_db)
-                    .collect { db ->
-                        def basename = db.getBaseName()
-                        def lineage = basename.contains('odb10') ? 'Y' : 'N'
-                        [[id: basename, lineage: lineage], db]
-                    }
-            }
-        }
-        else {
-            // Set BUSCO database to empty to allow for --auto-lineage
-            ch_db_for_busco = Channel
-                .of([[lineage: ''], []])
-                .collect()
         }
 
-        if (params.save_busco_db) {
-            // publish files downloaded by Busco
-            ch_downloads = BUSCO.out.busco_downloads
-                .groupTuple()
-                .map { _lin, downloads -> downloads[0] }
-                .toSortedList()
-                .flatten()
-            BUSCO_SAVE_DOWNLOAD(ch_downloads)
-
-            ch_versions = ch_versions.mix(BUSCO_SAVE_DOWNLOAD.out.versions.first())
-        }
-
-        BUSCO(ch_input_bins_for_qc, ch_db_for_busco)
-
-        BUSCO_SUMMARY(
-            BUSCO.out.summary_domain.collect { _meta, summary -> summary }.ifEmpty([]),
-            BUSCO.out.summary_specific.collect { _meta, summary -> summary }.ifEmpty([]),
-            BUSCO.out.failed_bin.collect { _meta, summary -> summary }.ifEmpty([]),
-        )
+        BUSCO_BUSCO(ch_bins, 'genome', busco_lineage, ch_busco_db, [], params.busco_clean)
 
+        qc_summaries = BUSCO_BUSCO.out.batch_summary
```

Comment:
Stylistic thing but should this be prefixed with `ch_`? Same with `busco_lineage`

Reply:
Yes, please use `ch_` for channels :)